services/src/main/java/com/google/daq/mqtt/mapping/MappingService.java

```diff
@@ -1,102 +1,126 @@
+          LOGGER.error("Registry Id not found for the message.");
+          return;
+        }
 
-  }
+        String discoveryNodeDeviceId = message.getAttributesOrDefault(DISCOVERY_NODE_DEVICE_ID_FIELD, "");
+        if (discoveryNodeDeviceId.isEmpty()) {
+          LOGGER.error("Discovery Node device Id not found for the message received.");
+        }
+        LOGGER.info("Starting Mapping process for registry: {}, family: {}, discoverNode deviceId: {}", registryId, mappingFamily, discoveryNodeDeviceId);
 
-  private void process() {
-    String timestamp = LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyyMMdd.HHmmss"));
+        SourceRepository repository = initRepository(registryId);
+        if (repository.clone(DEFAULT_TARGET_BRANCH)) {
+          String timestamp = LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyyMMdd.HHmmss"));
 
-    String exportBranch = String.format("proposal/%s/%s", SERVICE_NAME, timestamp);
-    if (!sourceRepository.checkoutNewBranch(exportBranch)) {
-      throw new RuntimeException("Unable to create and checkout export branch " + exportBranch);
-    }
+          String exportBranch = String.format("%s/%s/%s", TRIGGER_BRANCH, SERVICE_NAME, timestamp);
+          if (!repository.checkoutNewBranch(exportBranch)) {
+            throw new RuntimeException("Unable to create and checkout export branch " + exportBranch);
+          }
 
-    String udmiModelPath = sourceRepository.getUdmiModelPath();
-    (new Registrar()).processArgs(new ArrayList<>(List.of(udmiModelPath, projectSpec))).execute();
-    MappingAgent mappingAgent = new MappingAgent(new ArrayList<>(
-        List.of(udmiModelPath, projectSpec)));
-    mappingAgent.processMapping(new ArrayList<>(List.of(discoveryNodeDeviceId, mappingFamily)));
+          String udmiModelPath = repository.getUdmiModelPath();
+          (new Registrar()).processArgs(new ArrayList<>(List.of(udmiModelPath, projectSpec))).execute();
+          MappingAgent mappingAgent = new MappingAgent(new ArrayList<>(
+              List.of(udmiModelPath, projectSpec)));
+          mappingAgent.processMapping(new ArrayList<>(List.of(discoveryNodeDeviceId, mappingFamily)));
 
-    LOGGER.info("Committing and pushing changes to branch {}", exportBranch);
-    if (!sourceRepository.commitAndPush("Merge changes from source: MappingService")) {
-      throw new RuntimeException("Unable to commit and push changes to branch " + exportBranch);
+          LOGGER.info("Committing and pushing changes to branch {}", exportBranch);
+          if (!repository.commitAndPush("Merge changes from source: MappingService")) {
+            throw new RuntimeException("Unable to commit and push changes to branch " + exportBranch);
+          }
+          LOGGER.info("Export operation complete.");
+          repository.delete();
+        } else {
+          LOGGER.error("Could not clone repository! PR message was not published!");
+        }
+      }
+    } catch (IOException e) {
```

Comment:
try/catch is redundant with code in `AbstractPollingService` so I'd take it out. If there's something specific that's throwing the IOException I'd narrow down the try/catch to just that one thing and add some more meaningful detail? (If it's coming from the SourceRepsitory class I'd push it down into that?)